webapp/cas-server-webapp-config/src/main/java/org/apereo/cas/web/security/CasWebSecurityConfigurerAdapter.java

```diff
@@ -271,18 +272,25 @@ protected void configureEndpointAccessByIpAddress(final AuthorizeHttpRequestsCon
         requests
             .requestMatchers(endpoint)
             .access((authentication, context) -> {
-                val addresses = properties.getRequiredIpAddresses()
-                    .stream()
-                    .map(address -> '(' + address + ')')
-                    .collect(Collectors.joining("|"));
                 val remoteAddr = StringUtils.defaultIfBlank(
                     context.getRequest().getHeader(casProperties.getAudit().getEngine().getAlternateClientAddrHeaderName()),
                     context.getRequest().getRemoteAddr());
-                LOGGER.trace("Attempting to match [{}] against [{}] as a regular expression", remoteAddr, addresses);
-                val matcher = RegexUtils.createPattern(addresses, Pattern.CASE_INSENSITIVE).matcher(remoteAddr);
-                val granted = matcher.matches();
+
+                val addresses = properties.getRequiredIpAddresses()
+                        .stream()
+                        .filter(addr -> FunctionUtils.doAndHandle(() -> {
+                            val ipAddressMatcher = new IpAddressMatcher(addr);
+                            LOGGER.trace("Attempting to match [{}] against [{}] as a IP or netmask", remoteAddr, addr);
+                            return ipAddressMatcher.matches(remoteAddr);
+                        }, e -> {
+                            val matcher = RegexUtils.createPattern(addr, Pattern.CASE_INSENSITIVE).matcher(remoteAddr);
+                            LOGGER.trace("Attempting to match [{}] against [{}] as a regular expression", remoteAddr, addr);
+                            return matcher.matches();
```

Comment:
Excellent. LGTM. Final comment: should/could we use `find()` instead of `matches()`? I think the former is slightly more forgiving.

Reply:
Wouldn't using find() force people to do something like `^10.+` instead of `10.+` to avoid matching 192.168.10.0? I think most people would be doing a "starts with" match and if they forgot the ^ then they could have a rule that someone could bypass.

Reply:
Sounds good, sure.